### PR TITLE
Admission page

### PR DESCRIPTION
**Pull request by sathyaprabaS**

### CODE REVIEW

File: src/pageDrawers/AdmissionDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import Drawer from "@mui/material/Drawer";
+import { Box } from "@mui/system";
+import { Container, Divider, Typography } from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import Grid from "@mui/material/Grid";
+import { TextField } from "@mui/material";
+
+function AdmissionDrawer({ isOpen, onClose, dataItem }) {
+  if (!dataItem) {
```
Comment (praveensumathi) [inline src/pageDrawers/AdmissionDrawer.tsx:10]: change the name to selected item , and use the useState set the props to that state inside the useEffect

File: src/pageDrawers/AdmissionDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                    <Typography>Religion</Typography>
+                    <TextField
+                      disabled
+                      variant="outlined"
+                      fullWidth
+                      value={dataItem.religion}
+                    />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Typography>FatherName</Typography>
```
Comment (praveensumathi) [inline src/pageDrawers/AdmissionDrawer.tsx:124]: give gap father name

File: src/pages/home/Admission.tsx [hunk 3]
```diff
@@ -1,13 +1,137 @@
...
+              </TableCell>
+              <TableCell className={classes.textCenter}>
+                <Typography variant="subtitle1" fontWeight="bold">
+                  Actions
+                </Typography>
+              </TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {admisionData &&
```
Comment (praveensumathi) [inline src/pages/home/Admission.tsx:90]: check the length of the array

File: src/pages/home/Admission.tsx [hunk 4]
```diff
@@ -1,13 +1,137 @@
...
+              admisionData.map((dataItem, index) => (
```
Comment (praveensumathi) [inline src/pages/home/Admission.tsx:91]: change the dataItem to admission

File: src/pages/home/Admission.tsx [hunk 5]
```diff
@@ -1,13 +1,137 @@
...
+import { IAdmission } from "../../interface/types";
+import FormControlLabel from "@mui/material/FormControlLabel";
+import Checkbox from "@mui/material/Checkbox";
 
 function Admission() {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const [selectedDataItem, setSelectedDataItem] = useState<IAdmission | null>(
+    null
+  );
+  const [selectedDataId, setSelectedDataId] = useState(null);
```
Comment (praveensumathi) [inline src/pages/home/Admission.tsx:31]: remove this state

File: src/pages/home/Admission.tsx [hunk 6]
```diff
@@ -1,13 +1,137 @@
...
+
+  const classes = useCommonStyle();
+
+  const openDrawer = (dataItemId) => () => {
+    setSelectedDataId(dataItemId);
+    const selectedItem = admisionData.find((item) => item.id === dataItemId);
+    setSelectedDataItem(selectedItem || null);
```
Comment (praveensumathi) [inline src/pages/home/Admission.tsx:38]: should not set the null

File: src/pages/home/Admission.tsx [hunk 7]
```diff
@@ -1,13 +1,137 @@
...
+    setIsDrawerOpen(true);
+  };
+
+  const closeDrawer = () => {
+    setIsDrawerOpen(false);
```
Comment (praveensumathi) [inline src/pages/home/Admission.tsx:43]: set the selected item also null

File: src/pages/home/Admission.tsx [hunk 8]
```diff
@@ -1,13 +1,137 @@
...
+                  </TableCell>
+                  <TableCell>
+                    <Box
+                      sx={{
+                        cursor: "pointer",
+                        display: "flex",
+                        justifyContent: "space-around",
+                      }}
+                    >
+                      <IconButton onClick={openDrawer(dataItem.id)}>
```
Comment (praveensumathi) [inline src/pages/home/Admission.tsx:119]: sent the admission directly

File: src/pageDrawers/AdmissionDrawer.tsx (hunk 1 above)
Comment (sathyaprabaS) [reply src/pageDrawers/AdmissionDrawer.tsx:10]: changed

File: src/pageDrawers/AdmissionDrawer.tsx (hunk 2 above)
Comment (sathyaprabaS) [reply src/pageDrawers/AdmissionDrawer.tsx:124]: fixed

File: src/pages/home/Admission.tsx (hunk 4 above)
Comment (sathyaprabaS) [reply src/pages/home/Admission.tsx:91]: changed

File: src/pages/home/Admission.tsx (hunk 3 above)
Comment (sathyaprabaS) [reply src/pages/home/Admission.tsx:90]: checked

File: src/pages/home/Admission.tsx (hunk 6 above)
Comment (sathyaprabaS) [reply src/pages/home/Admission.tsx:38]: resolved

File: src/pages/home/Admission.tsx (hunk 5 above)
Comment (sathyaprabaS) [reply src/pages/home/Admission.tsx:31]: removed

File: src/pages/home/Admission.tsx (hunk 8 above)
Comment (sathyaprabaS) [reply src/pages/home/Admission.tsx:119]: changed

File: src/pages/home/Admission.tsx (hunk 7 above)
Comment (sathyaprabaS) [reply src/pages/home/Admission.tsx:42]: set

File: src/pageDrawers/AdmissionDrawer.tsx [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+import { Box } from "@mui/system";
+import { Container, Divider, Typography } from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import Grid from "@mui/material/Grid";
+import { TextField } from "@mui/material";
+
+interface IProps {
+  isOpen: boolean;
+  onClose: () => void;
+  selectedItem: any;
```
Comment (praveensumathi) [inline src/pageDrawers/AdmissionDrawer.tsx:12]: why this is any ?

File: src/pageDrawers/AdmissionDrawer.tsx [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+function AdmissionDrawer(props: IProps) {
+  const [displaySelectedData, setDisplaySelectedData] = useState(null);
```
Comment (praveensumathi) [inline src/pageDrawers/AdmissionDrawer.tsx:16]: change the "displaySelectedData" to "selectedAdmissionItem"

File: src/pageDrawers/AdmissionDrawer.tsx [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+                    <Typography>Caste</Typography>
+                    <TextField
+                      disabled
+                      variant="outlined"
+                      fullWidth
+                      value={selectedItem.caste}
+                    />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Typography>Natinoality</Typography>
```
Comment (praveensumathi) [inline src/pageDrawers/AdmissionDrawer.tsx:105]: check the "Natinoality" spelling

File: src/pageDrawers/AdmissionDrawer.tsx (hunk 10 above)
Comment (sathyaprabaS) [reply src/pageDrawers/AdmissionDrawer.tsx:16]: changed

File: src/pageDrawers/AdmissionDrawer.tsx (hunk 11 above)
Comment (sathyaprabaS) [reply src/pageDrawers/AdmissionDrawer.tsx:105]: changed

**Review by praveensumathi — APPROVED** [review]

**Pull request merged**